cli/commands/scaffold/action.go

```diff
@@ -221,62 +221,93 @@ func Run(ctx context.Context, opts *options.TerragruntOptions, moduleURL, templa
 	return nil
 }
 
-// prepareBoilerplateFiles prepares boilerplate files.
-func prepareBoilerplateFiles(ctx context.Context, opts *options.TerragruntOptions, templateURL string, tempDir string) (string, error) {
-	// identify template url
-	templateDir := ""
+// generateDefaultTemplate - write default template to provided dir
+func generateDefaultTemplate(boilerplateDir string) (string, error) {
+	const ownerWriteGlobalReadPerms = 0644
+	if err := os.WriteFile(util.JoinPath(boilerplateDir, "terragrunt.hcl"), []byte(DefaultTerragruntTemplate), ownerWriteGlobalReadPerms); err != nil {
+		return "", errors.New(err)
+	}
 
-	if templateURL != "" {
-		// process template url if was passed
-		parsedTemplateURL, err := tf.ToSourceURL(templateURL, tempDir)
-		if err != nil {
-			return "", errors.New(err)
-		}
+	if err := os.WriteFile(util.JoinPath(boilerplateDir, "boilerplate.yml"), []byte(DefaultBoilerplateConfig), ownerWriteGlobalReadPerms); err != nil {
+		return "", errors.New(err)
+	}
 
-		parsedTemplateURL, err = rewriteTemplateURL(ctx, opts, parsedTemplateURL)
-		if err != nil {
-			return "", errors.New(err)
-		}
-		// regenerate template url with all changes
-		templateURL = parsedTemplateURL.String()
+	return boilerplateDir, nil
+}
 
-		// prepare temporary directory for template
-		templateDir, err = os.MkdirTemp("", "template")
-		if err != nil {
-			return "", errors.New(err)
-		}
+// parseTemplateURL - parse URL and download files
+func parseTemplateURL(ctx context.Context, opts *options.TerragruntOptions, templateURL string, tempDir string) (string, error) {
```

Comment:
I think it's better to call this something like `downloadTemplate`, as we're not just parsing the URL, but also getting it.

Reply:
Updated